hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1228,17 +1238,61 @@ public void restart() throws IOException {
       // Allow OM to start as Http Server failure is not fatal.
       LOG.error("OM HttpServer failed to start.", ex);
     }
-
     omRpcServer.start();
+
     isOmRpcServerRunning = true;
 
+    if (isLeader()) {
```

Comment:
this check should be inside run method in TrashPolicyOzone. I also do not see a problem if it was run on all OM nodes.